packages/beacon-node/src/monitoring/system.ts

```diff
@@ -5,264 +5,211 @@ import {Logger} from "@lodestar/utils";
 
 type MiscOs = "lin" | "win" | "mac" | "unk";
 
-// static data only needs to be collected once
-let staticDataCollected = false;
-// disk I/O is not measurable in some environments
-let diskIOMeasurable = true;
-
-let cpuCores = 0;
-let cpuThreads = 0;
-let cpuNodeSystemSecondsTotal = 0;
-let cpuNodeUserSecondsTotal = 0;
-let cpuNodeIdleSecondsTotal = 0;
-let memoryNodeBytesTotal = 0;
-let memoryNodeBytesFree = 0;
-let memoryNodeBytesCached = 0;
-let memoryNodeBytesBuffers = 0;
-let diskNodeBytesTotal = 0;
-let diskNodeBytesFree = 0;
-let diskNodeReadsTotal = 0;
-let diskNodeWritesTotal = 0;
-let networkNodeBytesTotalReceive = 0;
-let networkNodeBytesTotalTransmit = 0;
-let miscNodeBootTsSeconds = 0;
-let miscOs: MiscOs = "unk";
-
-/**
- * Collect system data and update cached values
- */
-export async function collectSystemData(logger: Logger): Promise<void> {
-  const debug = (dataType: string, e: Error): void => logger.debug(`Failed to collect ${dataType} data`, {}, e);
-
-  await Promise.all([
-    collectStaticData().catch((e) => debug("static system", e)),
-    collectCpuData().catch((e) => debug("CPU", e)),
-    collectMemoryData().catch((e) => debug("memory", e)),
-    collectDiskData().catch((e) => debug("disk", e)),
-    collectNetworkData().catch((e) => debug("network", e)),
-  ]);
-
-  miscNodeBootTsSeconds = getSystemBootTime();
-}
+class System {
+  // static data only needs to be collected once
+  private staticDataCollected = false;
+  // disk I/O is not measurable in some environments
+  private diskIOMeasurable = true;
+
+  private _cpuCores = 0;
```

Comment:
I generally would not prefix private properties with `_` but needs to be done in that case as we also have a `get` accessor for each property which uses the same name but without `_`.

I saw this convention used in other places as well so I assume it should be fine to use here

https://github.com/ChainSafe/lodestar/blob/b64a67c71513e5d8b0ff2b8479ee90ed84da94d4/packages/beacon-node/src/chain/clock/LocalClock.ts#L20-L21

Reply:
why do we need private properties and getters? Rather than public properties (or public readonly properties).

Reply:
I would kinda be fine with public readonly properties as well but I cannot make them readonly as they are modified within the class itself.

https://github.com/ChainSafe/lodestar/blob/0cd2d205744557d9ab67e8f1d631c2785c0efe7d/packages/beacon-node/src/monitoring/system.ts#L46

But even if I can make them readonly I still think it is not ideal in most cases, I would rather prefer to have explicit getters and setters to retrieve/modify properties of a class and make them private by default at the cost of a bit more code/verbosity.

Also the code below should never be allowed, that is why I can only use private properties + getters

![image](https://user-images.githubusercontent.com/38436224/221689259-872c3c55-8238-49bd-8295-d270d3440270.png)

Reply:
The `readonly` only works for the TS environment. Once compiled we didn't had that function anymore. This code is not for external use so its fine to use readonly.

Reply:
Malicious user could still corrupt the values like `(system as any)._cpuCores = 0`.
Or recreate new instances like `new Object.getPrototypeOf(system).constructor()`.
The general assumption we're currently using in the project: internal lodestar code is running in a trusted environment. You don't need to have real data hiding at runtime, you don't need to build protections against malicious misuse. No need to protect against "What if someone sets this module's private counter to zero?"

(If we do end up supporting running lodestar in an untrusted environment, we would wrap the entirety and expose only a controlled api that doesn't allow access to internals)

So here, I would use some judgement about how we expect this module to be used, as an internal datasource for the monitoring service. My feeling is that this PR should really be focusing on matters of maintainability of the code. How easy / hard am I making our lives when we inevitably come back in 6 months or a year and need to add / delete / etc system data to expose to the monitoring api?

Reply:
@wemeetagain It is not really about a malicious user but just keeping the external API as simple/limited as possible but you are probably right, since the use case of this is currently limited to as an internal datasource for the monitoring service, we should be fine with public modifiable properties as well.

If we go with that approach I think changes done in [this commit](https://github.com/ChainSafe/lodestar/pull/5202/commits/45c8cfd5aea26ca7123a1157843a79eaf5f525a6) also do not make sense, as the class is not exported and can therefore not be initialized twice.